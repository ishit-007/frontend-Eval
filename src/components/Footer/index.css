.footer {
  height: 10vh;
  margin-top: 5rem;
  background-color: black;
  color: white;
  display: flex;
  justify-content: space-around;
  font-weight: 600;
  font-size: larger;
  align-items: center;
}
.footer button {
  padding: 5px;
  height: 40%;
  font-weight: 600;
  border: none;
  border-radius: 15px;
}
.themes {
  display: flex;
  gap: 1rem;
  height: 100%;
  align-items: center;
}
