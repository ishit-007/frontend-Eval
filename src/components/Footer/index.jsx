import './index.css';

import React from 'react';
import { makeRequest, GET_THEMES_DATA } from '../../util/makeRequest';

const Footer = () => {
  const [themes, setThemes] = React.useState([]);
  React.useEffect(() => {
    makeRequest(GET_THEMES_DATA, {}).then(themesData => {
      setThemes(themesData.themes);
    });
  }, []);
  return (
    <div className="footer">
      <div className="themes">
        <p>THEMES</p>
        {themes.map(theme => {
          return (
            <div
              className="theme"
              key={theme.id}
              style={{
                backgroundColor: theme.colorHexCode,
                height: '60px',
                width: '60px',
                borderRadius: '10px',
              }}></div>
          );
        })}
      </div>

      <button>SAVE THEME</button>
    </div>
  );
};

export default Footer;
