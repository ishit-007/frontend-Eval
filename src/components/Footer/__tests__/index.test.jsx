import { render } from '@testing-library/react';
import Footer from '..';
import renderer from 'react-test-renderer';
import React from 'react';

describe('NavBar', () => {
  it('should render the Footer component', async () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
