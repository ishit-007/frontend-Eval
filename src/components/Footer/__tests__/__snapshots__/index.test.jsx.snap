// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render the Footer component 1`] = `
<div
  className="footer"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <div
    className="themes"
  >
    <p>
      THEMES
    </p>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "color": undefined,
      }
    }
  >
    SAVE THEME
  </button>
</div>
`;
